core/src/main/kotlin/fr/sncf/osrd/stdcm/graph/STDCMNode.kt

```diff
@@ -20,21 +20,21 @@ data class STDCMNode(
             Block
         >?, // Position on a block, if this node isn't on the transition between blocks (stop)
     val stopDuration: Double?, // When the node is a stop, how long the train remains here
-    var remainingTimeEstimation: Double =
-        0.0, // Estimation of the min time it takes to reach the end from this node
+    var remainingTimeEstimation:
+        Double?, // Estimation of the min time it takes to reach the end from this node
 ) : Comparable<STDCMNode> {
 
     override fun compareTo(other: STDCMNode): Int {
         val currentRunningTime = time - totalPrevAddedDelay
         val otherCurrentRunningTime = other.time - other.totalPrevAddedDelay
         return if (
             abs(
-                (currentRunningTime + remainingTimeEstimation) -
-                    (otherCurrentRunningTime + other.remainingTimeEstimation)
+                (currentRunningTime + (remainingTimeEstimation ?: 0.0)) -
```

Comment:
I kinda agreed with the comment, but in the end, I don't know if putting no default value and initializing `remainingTimeEstimation` is such a good idea. We often use `compareTo` or pass through the `if` condition in `findPathImpl()` before the value is initialized, which means we constantly need to reset it to 0.0 if null, which is basically the same as having a default value 0.0.

Reply:
In what scenario do we call `compareTo` without it being initialized? It shouldn't happen and could cause issues, it sounds like a case where we do want to crash.

If there's a legitimate reason for it to happen we can keep the 0, but for now it just seems really suspicious

Reply:
So I checked when it happens, it's when we don't set the heuristic. Which is indeed optional.

There's two options:

1. Have a default value `x -> 0.0` for the heuristic
2. Use 0.0 as default value in the node

Option 1 is a little more work, but 2 could fail silently if we mess something up